### PR TITLE
Astro deployment airflow upgrade no op 3055

### PR DESCRIPTION
**Pull request by jarfgit**
## Description
Provides an error message with more context to the user when they attempt to upgrade to the version of airflow they're currently running. 

To do the above I ended up moving a small amount of code that I don't think was covered by unit tests. So I updated / added unit tests as well. 

**Update:** Took a closer look and fixed some test data. The change below was totally unnecessary.
~NOTE: I'm not 100% sure that this is the change I want to make here to satisfy the tests:~
image

~However the applicable test was mocked with an 'updateDeploymentAirflow' object returned from the API and on line 266 of deployment.go we're trying to access r.Data.UpdateDeploymentAirflow, so I'm flipping a coin 😄~

## 🎟 Issue(s)

Resolves astronomer/issues#3055

## 🧪 Functional Testing
make test

## 📸 Screenshots
N/A

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [x] Added/updated applicable tests
- [x] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [x] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [x] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: deployment/deployment.go [hunk 1]
```diff
@@ -372,6 +371,22 @@ func getDeployment(deploymentId string, client *houston.Client) (*houston.Deploy
 	return &r.Data.GetDeployment, nil
 }
 
+func getUpdateDeploymentAirflow(deploymentId string, client *houston.Client) (*houston.Deployment, error) {
```
Comment (bote795) [inline deployment/deployment.go:374]: whats the difference between this function and getDeployment?

Comment (jarfgit) [reply deployment/deployment.go]: Nothing :P 

I took a closer look and now understand the issue was actually mocked API data missing from the tests. I removed the unnecessary extra stuff that I added. This PR should be a bit more straightforward now.

**Review by bote795 — APPROVED** [review]
lgtm

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
 All notable changes to this project will be documented in this file.
+## [0.23.2] - 2020-06-23
```
Comment (andriisoldatenko) [inline CHANGELOG.md:4]: wrong year? also wrong version.

File: deployment/deployment_test.go [hunk 3]
```diff
@@ -529,6 +534,78 @@ func TestAirflowUpgradeCancelError(t *testing.T) {
 	assert.Error(t, err, "API error (500):")
 }
 
+func TestAirflowUpgradeEmptyDesiredVersion(t *testing.T) {
+	testUtil.InitTestConfig()
+	okResponse := `{"data": {
```
Comment (andriisoldatenko) [inline deployment/deployment_test.go:539]: can you please fix formating of json using 2 spaces and remove tabs?

File: CHANGELOG.md (hunk 2 above)
Comment (jarfgit) [reply CHANGELOG.md:4]: Whoops, good catch. Should this be a minor version bump to 0.24.0?

File: deployment/deployment_test.go (hunk 3 above)
Comment (jarfgit) [reply deployment/deployment_test.go]: Done.

File: deployment/deployment.go [hunk 4]
```diff
@@ -385,19 +384,26 @@ func meetsAirflowUpgradeReqs(airflowVersion string, desiredAirflowVersion string
 		return err
 	}
 
+	currentVersion, err := semver.NewVersion(airflowVersion)
+	if err != nil {
+		return err
```
Comment (andriisoldatenko) [inline deployment/deployment.go:389]: it's easy to test with test based on invalid airflowVersion e.g. "-3.2.5"

**Review by andriisoldatenko — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
 All notable changes to this project will be documented in this file.
+## [0.24.0] - 2021-06-23
```
Comment (andriisoldatenko) [inline CHANGELOG.md:4]:
```suggestion
## [0.25.2] - 2021-06-23
```

**Pull request merged**